src/base/compiler.hh

```diff
@@ -90,7 +90,7 @@
 // This version is for macros which are statement-like, which frequently use
 // "do {} while (0)" to make their syntax look more like normal c++ statements.
 #  define GEM5_DEPRECATED_MACRO_STMT(name, definition, message) \
-     do {{definition;} GEM5_DEPRECATED_MACRO(name, {}, message);} while (0)
```

Comment:
Looks like GCC is not happy with this change, will try to fix (hopefully `(void)0` works).

Reply:
Fixed now.